src-java/kilda-model/src/main/java/org/openkilda/model/Cookie.java

```diff
@@ -45,101 +47,110 @@
-    public static final long VERIFICATION_UNICAST_VXLAN_RULE_COOKIE     = 0x07L | DEFAULT_RULE_FLAG;
-    public static final long MULTITABLE_PRE_INGRESS_PASS_THROUGH_COOKIE = 0x08L | DEFAULT_RULE_FLAG;
-    public static final long MULTITABLE_INGRESS_DROP_COOKIE             = 0x09L | DEFAULT_RULE_FLAG;
-    public static final long MULTITABLE_POST_INGRESS_DROP_COOKIE        = 0x0AL | DEFAULT_RULE_FLAG;
-    public static final long MULTITABLE_EGRESS_PASS_THROUGH_COOKIE      = 0x0BL | DEFAULT_RULE_FLAG;
-    public static final long MULTITABLE_TRANSIT_DROP_COOKIE             = 0x0CL | DEFAULT_RULE_FLAG;
-    public static final long LLDP_INPUT_PRE_DROP_COOKIE                 = 0x0DL | DEFAULT_RULE_FLAG;
-    public static final long LLDP_TRANSIT_COOKIE                        = 0x0EL | DEFAULT_RULE_FLAG;
-    public static final long LLDP_INGRESS_COOKIE                        = 0x0FL | DEFAULT_RULE_FLAG;
-    public static final long LLDP_POST_INGRESS_COOKIE                   = 0x10L | DEFAULT_RULE_FLAG;
-    public static final long LLDP_POST_INGRESS_VXLAN_COOKIE             = 0x11L | DEFAULT_RULE_FLAG;
-    public static final long LLDP_POST_INGRESS_ONE_SWITCH_COOKIE        = 0x12L | DEFAULT_RULE_FLAG;
-    public static final long ARP_INPUT_PRE_DROP_COOKIE                  = 0x13L | DEFAULT_RULE_FLAG;
-    public static final long ARP_TRANSIT_COOKIE                         = 0x14L | DEFAULT_RULE_FLAG;
-    public static final long ARP_INGRESS_COOKIE                         = 0x15L | DEFAULT_RULE_FLAG;
-    public static final long ARP_POST_INGRESS_COOKIE                    = 0x16L | DEFAULT_RULE_FLAG;
-    public static final long ARP_POST_INGRESS_VXLAN_COOKIE              = 0x17L | DEFAULT_RULE_FLAG;
-    public static final long ARP_POST_INGRESS_ONE_SWITCH_COOKIE         = 0x18L | DEFAULT_RULE_FLAG;
-
-    // 9 bits cookie type "field"
-    public static final long TYPE_MASK                               = 0x1FF0_0000_0000_0000L;
-    public static final long FLOW_COOKIE_TYPE                        = 0x0000_0000_0000_0000L;
-    public static final long LLDP_INPUT_CUSTOMER_TYPE                = 0x0010_0000_0000_0000L;
-    public static final long MULTITABLE_ISL_VLAN_EGRESS_RULES_TYPE   = 0x0020_0000_0000_0000L;
-    public static final long MULTITABLE_ISL_VXLAN_EGRESS_RULES_TYPE  = 0x0030_0000_0000_0000L;
-    public static final long MULTITABLE_ISL_VXLAN_TRANSIT_RULES_TYPE = 0x0040_0000_0000_0000L;
-    public static final long MULTITABLE_INGRESS_RULES_TYPE           = 0x0050_0000_0000_0000L;
-    public static final long ARP_INPUT_CUSTOMER_TYPE                 = 0x0060_0000_0000_0000L;
-
-    private final long value;
-
-    /**
-     * Create {@code Cookie} instance and perform it's validation.
-     */
-    public static Cookie decode(long rawValue) {
-        Cookie cookie = new Cookie(rawValue);
-        cookie.ensureNoFlagsConflicts();
-        return cookie;
-    }
+    // FIXME(surabujin): get rid from this constants
```

Comment:
Honestly I don't like the idea of creating so many classes (schemas, types, etc).
`Cookie` class is a one place were you can find all information related to cookie(types, subtypes, schema with description of all bits used in cookie)
Information from the left(which were removed ) is readable, complete and easy understandable. 

Now we have several classes, with several subtypes, placed in different files. It's hard to work with such number of classes. It's easy to make a mistake. Usage in code doesn't looks like a syntax sugar. Example

Was:
```
long cookie = flowResources.getUnmaskedCookie();
Cookie.buildForwardCookie(cookie)
```
Now:
```
Cookie blank = FlowSegmentCookieSchema.INSTANCE.makeBlank();
Cookie flowCookie = FlowSegmentCookieSchema.INSTANCE.setFlowEffectiveId(
blank, flowResources.getUnmaskedCookie());
FlowSegmentCookieSchema.INSTANCE.setDirection(flowCookie, FlowPathDirection.FORWARD)
```

Maybe we shouldn't make simple things difficult?

Reply:
At first, we have used `long` type to represent cookie in our code. It produced a great number of different errors in all imaginable places. So it was wrapped into special type to ensure that if code expect the cookie it will receive the cookie and compiler will guarantee this - this is when Cookie class appears. At some later point, this guarantee was eliminated with methods like 
```
    public static long encodeIslVlanEgress(int port) {
        return new Cookie(port | DEFAULT_RULE_FLAG)
                .setType(CookieType.MULTI_TABLE_ISL_VLAN_EGRESS_RULES)
                .getValue();
    }
```
i.e. methods that receive simple/generic type and return simple/generic type. 

Later cookie "schema" becomes more complex - type field was introduced and this field defines the meaning of other bits. So now cookie have not documented and not defined in any way restrictions which bits can be defined at the same time and which must collide. Keeping this all in "simple" way how it now - is keeping it in the way this restriction can't be easily defined in code and as the result can't be covered with tests.

The goal of all these "extra classes as you say" is to define each "conflicting" schema in its own class. It is verbose enough way to define "compatible" bits (and not compatible). Another goal is to eliminate the possibility to use simple/generic types instead of Cookie type (but it can't be reached so easy).